# **Exploratory Data Analysis on the Shark Tank India Season 1**
---


In this notebook, we'll look at the show's statistics to answer some interesting questions regarding the sharks, transactions, and firms.


First, we'll examine the likelihood that a settlement will be reached. Next, we'll find out which shark is most likely to **invest in** a deal and which sector of the economy the sharks are most interested in.


Here, we begin by importing the necessary Python libraries.

In [49]:
import numpy as np

import pandas as pd

import matplotlib.pyplot as plt

import seaborn as sns

import plotly.express as px

Dataset link :
https://www.kaggle.com/datasets/shivavashishtha/shark-tank-india-dataset/code

Load the dataset
---

In [50]:
df=pd.read_csv('Shark Tank India Dataset.csv')

df.head()

episode_number  pitch_number           brand_name  \
0               1             1  BluePine Industries   
1               1             2        Booz scooters   
2               1             3  Heart up my Sleeves   
3               2             4           Tagz Foods   
4               2             5       Head and Heart   

                                            idea  deal  pitcher_ask_amount  \
0                                   Frozen Momos     1                50.0   
1  Renting e-bike for mobility in private spaces     1                40.0   
2                             Detachable Sleeves     1                25.0   
3                           Healthy Potato Chips     1                70.0   
4                       Brain Development Course     0                50.0   

   ask_equity  ask_valuation  deal_amount  deal_equity  ...  ashneer_deal  \
0         5.0        1000.00         75.0        16.00  ...             1   
1        15.0         266.67         40.0        50.00  ...             1   
2        10.0         250.00         25.0        30.00  ...             0   
3         1.0        7000.00         70.0         2.75  ...             1   
4         5.0        1000.00          0.0         0.00  ...             0   

   anupam_deal  aman_deal  namita_deal  vineeta_deal  peyush_deal  \
0            0          1            0             1            0   
1            0          0            0             1            0   
2            1          0            0             1            0   
3            0          0            0             0            0   
4            0          0            0             0            0   

   ghazal_deal  total_sharks_invested  amount_per_shark  equity_per_shark  
0            0                      3              25.0          5.333333  
1            0                      2              20.0         25.000000  
2            0                      2              12.5         15.000000  
3            0                      1              70.0          2.750000  
4            0                      0               0.0          0.000000  

[5 rows x 28 columns]

In [51]:
df.shape

(117, 28)

In [52]:
df.describe()

episode_number  pitch_number        deal  pitcher_ask_amount  \
count      117.000000    117.000000  117.000000          117.000000   
mean        18.735043     59.000000    0.555556          319.854709   
std         10.070778     33.919021    0.499041         2767.842777   
min          1.000000      1.000000    0.000000            0.001010   
25%         10.000000     30.000000    0.000000           45.000000   
50%         19.000000     59.000000    1.000000           50.000000   
75%         27.000000     88.000000    1.000000           80.000000   
max         35.000000    117.000000    1.000000        30000.000000   

       ask_equity  ask_valuation  deal_amount  deal_equity  deal_valuation  \
count  117.000000     117.000000   117.000000   117.000000      117.000000   
mean     5.188034    3852.462479    31.982915     8.963504      467.104872   
std      3.892121   11931.601957    36.687391    13.106769      919.988864   
min      0.250000       0.010000     0.000000     0.000000        0.000000   
25%      2.500000     666.670000     0.000000     0.000000        0.000000   
50%      5.000000    1250.000000    25.000000     3.000000      100.000000   
75%      7.500000    2857.140000    50.000000    15.000000      500.000000   
max     25.000000  120000.000000   150.000000    75.000000     6666.670000   

       ashneer_present  ...  ashneer_deal  anupam_deal   aman_deal  \
count       117.000000  ...    117.000000   117.000000  117.000000   
mean          0.837607  ...      0.179487     0.205128    0.239316   
std           0.370397  ...      0.385410     0.405532    0.428501   
min           0.000000  ...      0.000000     0.000000    0.000000   
25%           1.000000  ...      0.000000     0.000000    0.000000   
50%           1.000000  ...      0.000000     0.000000    0.000000   
75%           1.000000  ...      0.000000     0.000000    0.000000   
max           1.000000  ...      1.000000     1.000000    1.000000   

       namita_deal  vineeta_deal  peyush_deal  ghazal_deal  \
count   117.000000    117.000000   117.000000   117.000000   
mean      0.188034      0.128205     0.230769     0.059829   
std       0.392420      0.335756     0.423137     0.238190   
min       0.000000      0.000000     0.000000     0.000000   
25%       0.000000      0.000000     0.000000     0.000000   
50%       0.000000      0.000000     0.000000     0.000000   
75%       0.000000      0.000000     0.000000     0.000000   
max       1.000000      1.000000     1.000000     1.000000   

       total_sharks_invested  amount_per_shark  equity_per_shark  
count             117.000000        117.000000        117.000000  
mean                1.230769         18.132481          5.583590  
std                 1.410457         23.588682         10.803799  
min                 0.000000          0.000000          0.000000  
25%                 0.000000          0.000000          0.000000  
50%                 1.000000         10.000000          1.250000  
75%                 2.000000         25.000000          6.000000  
max                 5.000000        100.000000         75.000000  

[8 rows x 26 columns]

In [53]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117 entries, 0 to 116
Data columns (total 28 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   episode_number         117 non-null    int64  
 1   pitch_number           117 non-null    int64  
 2   brand_name             117 non-null    object 
 3   idea                   117 non-null    object 
 4   deal                   117 non-null    int64  
 5   pitcher_ask_amount     117 non-null    float64
 6   ask_equity             117 non-null    float64
 7   ask_valuation          117 non-null    float64
 8   deal_amount            117 non-null    float64
 9   deal_equity            117 non-null    float64
 10  deal_valuation         117 non-null    float64
 11  ashneer_present        117 non-null    int64  
 12  anupam_present         117 non-null    int64  
 13  aman_present           117 non-null    int64  
 14  namita_present         117 non-null    int64  
 15  vineet

Data cleaning
--

In [54]:
df.isnull().sum()

episode_number           0
pitch_number             0
brand_name               0
idea                     0
deal                     0
pitcher_ask_amount       0
ask_equity               0
ask_valuation            0
deal_amount              0
deal_equity              0
deal_valuation           0
ashneer_present          0
anupam_present           0
aman_present             0
namita_present           0
vineeta_present          0
peyush_present           0
ghazal_present           0
ashneer_deal             0
anupam_deal              0
aman_deal                0
namita_deal              0
vineeta_deal             0
peyush_deal              0
ghazal_deal              0
total_sharks_invested    0
amount_per_shark         0
equity_per_shark         0
dtype: int64

Data suggests that the company has not received an offer, hence there is no transaction where a rejected offer is equivalent to na. As a result, we are able to make that offer and reject it.

In [55]:
df.isna().any()

episode_number           False
pitch_number             False
brand_name               False
idea                     False
deal                     False
pitcher_ask_amount       False
ask_equity               False
ask_valuation            False
deal_amount              False
deal_equity              False
deal_valuation           False
ashneer_present          False
anupam_present           False
aman_present             False
namita_present           False
vineeta_present          False
peyush_present           False
ghazal_present           False
ashneer_deal             False
anupam_deal              False
aman_deal                False
namita_deal              False
vineeta_deal             False
peyush_deal              False
ghazal_deal              False
total_sharks_invested    False
amount_per_shark         False
equity_per_shark         False
dtype: bool

In [56]:
df.columns

Index(['episode_number', 'pitch_number', 'brand_name', 'idea', 'deal',
       'pitcher_ask_amount', 'ask_equity', 'ask_valuation', 'deal_amount',
       'deal_equity', 'deal_valuation', 'ashneer_present', 'anupam_present',
       'aman_present', 'namita_present', 'vineeta_present', 'peyush_present',
       'ghazal_present', 'ashneer_deal', 'anupam_deal', 'aman_deal',
       'namita_deal', 'vineeta_deal', 'peyush_deal', 'ghazal_deal',
       'total_sharks_invested', 'amount_per_shark', 'equity_per_shark'],
      dtype='object')

Percentage of comapnies got investment or not?
---

In [57]:
got_in = df[df['deal_amount'] > 0.0].shape[0]

not_in = df[df['deal_amount'] == 0.0].shape[0]


labels = ["Got Investment", "Not Received Investment"]

values = [got_in, not_in]

# Define custom colors

colors = ['#3498db', '#e74c3c']

fig = px.pie(values=values, names=labels, hole=0.4, color_discrete_sequence=colors)

fig.update_traces(textposition='inside', textinfo='percent+label')


fig.update_layout(title="Number of Companies which Got Investment or Not -", showlegend=False)

fig.show()

In [58]:
cleaned_df = df[df["brand_name"] != "Gopal's 56"]

In [59]:
cleaned_df.head()

episode_number  pitch_number           brand_name  \
0               1             1  BluePine Industries   
1               1             2        Booz scooters   
2               1             3  Heart up my Sleeves   
3               2             4           Tagz Foods   
4               2             5       Head and Heart   

                                            idea  deal  pitcher_ask_amount  \
0                                   Frozen Momos     1                50.0   
1  Renting e-bike for mobility in private spaces     1                40.0   
2                             Detachable Sleeves     1                25.0   
3                           Healthy Potato Chips     1                70.0   
4                       Brain Development Course     0                50.0   

   ask_equity  ask_valuation  deal_amount  deal_equity  ...  ashneer_deal  \
0         5.0        1000.00         75.0        16.00  ...             1   
1        15.0         266.67         40.0        50.00  ...             1   
2        10.0         250.00         25.0        30.00  ...             0   
3         1.0        7000.00         70.0         2.75  ...             1   
4         5.0        1000.00          0.0         0.00  ...             0   

   anupam_deal  aman_deal  namita_deal  vineeta_deal  peyush_deal  \
0            0          1            0             1            0   
1            0          0            0             1            0   
2            1          0            0             1            0   
3            0          0            0             0            0   
4            0          0            0             0            0   

   ghazal_deal  total_sharks_invested  amount_per_shark  equity_per_shark  
0            0                      3              25.0          5.333333  
1            0                      2              20.0         25.000000  
2            0                      2              12.5         15.000000  
3            0                      1              70.0          2.750000  
4            0                      0               0.0          0.000000  

[5 rows x 28 columns]

In [60]:
cleaned_df.to_csv("cleaned_st.csv", index=False)


Maximum Pitch Request Amount
---



In [61]:
high = df[df["pitcher_ask_amount"] > 100]

figure = px.bar(
    high,
    
    x='brand_name',
    
    y='pitcher_ask_amount',
    
    title="Maximum Pitch Request Amount :",
    text_auto=True,
    
    color='deal_amount',
    
    color_continuous_scale='greens',  # You can choose other color scales, like 'greens', 'reds', etc.
    
    template="plotly_dark")

figure.update_layout(coloraxis_colorbar=dict(title='Deal Amount'))

figure.show()

Raise and Handle Equity
---

In [62]:
figure = px.bar(
    high,
    
    x='brand_name',
    
    y='ask_equity',
    
    title="Raise and Handle Equity :",
    
    text_auto=True,
    
    color='deal_equity',
    
    color_continuous_scale='greens',  # Use 'greens' for a white and green color scale
    
    template="plotly_dark")

figure.update_layout(coloraxis_colorbar=dict(title='Deal Equity'))

figure.show()

The blue colour map, which indicates that no deal occurred with the exception of "Aas Vidyalaya," made it evident that the majority of the brands requesting large sums of money had not obtained investment from the shark.


---
Minimal Pitch Amount Inquired
---

In [63]:
low=df[df["pitcher_ask_amount"]<20]
figure=px.bar(
    
    low,
    
    x='brand_name',
    
    y='pitcher_ask_amount',
    
    title="Lowest Pitch Ask Amount :",
    
    text_auto=True,
    
    color='deal_amount',
    
    color_continuous_scale='greens',  # Use 'greens' for a white and green color scale
    
    template="plotly_dark")

figure.show()

Out of all the brands that made the lowest pitch amount, only "Cocofit" and "Watt Technovations" received funding.

Cocofit and Watt Technovations had the lowest pitch amount for which sharks made a deal.


Number of brands in which sharks have invested.

---

In [64]:
num_deal_shark = [df.ashneer_deal.sum(), df.anupam_deal.sum(), df.aman_deal.sum(), df.namita_deal.sum(), df.vineeta_deal.sum(), df.peyush_deal.sum(), df.ghazal_deal.sum()]

all_sharks = ["Ashneer", "Anupam", "Aman", "Namita", "Vineeta", "Peyush", "Ghazal"]

# Define a custom color scale with unique colors for each shark

custom_colors = px.colors.qualitative.Set2

figure = px.bar(
    
    x=all_sharks,
    
    y=num_deal_shark,
    
    title="Number of deals done by individual shark :",
    
    color=all_sharks,  # Each bar will be assigned a unique color
    
    text_auto=True,
   
    template="plotly_dark",
    
    color_discrete_sequence=custom_colors

)

figure.show()


In [65]:
total_shark = 42

In [66]:
# Incorrect
print(total_shark)

# Correct
total_shark = 42

print(total_shark)


42
42


More than 50% of the shark's equity in which brand
---

In [67]:
df_equity = df[df['equity_per_shark'] >= 50]

figure = px.bar(
    
    df_equity,
    
    x='brand_name',
    
    y='equity_per_shark',
    
    title="More than 50% of the shark's equity in which brand",
    
    text_auto=True,
    
    color='deal_amount',
    
    template="plotly") # Change template from "plotly_dark" to "plotly" for a lighter background

figure.show()

Entrepreneurs' equity offered to investors in transactions
---

In [68]:
average_equity = df['deal_equity'].mean()

print(f"On Shark Tank India, entrepreneurs typically offer investors stock that is: {average_equity:.2f}")

On Shark Tank India, entrepreneurs typically offer investors stock that is: 8.96


The average equity offered by entrepreneurs to investors on Shark Tank India is: 8.96
